addons/addon-base-raas/packages/base-raas-cfn-templates/src/templates/onboard-account.cfn.yml

```diff
@@ -409,6 +405,110 @@ Resources:
       AliasName: !Join ['', ['alias/', Ref: Namespace, '-encryption-key']]
       TargetKeyId: !Ref EncryptionKey
 
+  CloudTrailBucket: 
+    DeletionPolicy: Retain
+    Type: AWS::S3::Bucket
+    Properties: 
+      BucketName: !Join ['-', ['swb', 'cloudtrail-log-bucket', Ref: AWS::AccountId]]
+      BucketEncryption:
+        ServerSideEncryptionConfiguration:
+          - ServerSideEncryptionByDefault:
+              SSEAlgorithm: AES256
+      PublicAccessBlockConfiguration:
+        BlockPublicAcls: true
+        BlockPublicPolicy: true
+        IgnorePublicAcls: true
+        RestrictPublicBuckets: true
+
+  CloudTrailBucketPolicy: 
+    Type: AWS::S3::BucketPolicy
+    Properties: 
+      Bucket: 
+        Ref: CloudTrailBucket
+      PolicyDocument: 
+        Version: "2012-10-17"
+        Statement: 
+          - 
+            Sid: "AWSCloudTrailAclCheck"
+            Effect: "Allow"
+            Principal: 
+              Service: "cloudtrail.amazonaws.com"
+            Action: "s3:GetBucketAcl"
+            Resource: 
+              !Sub |-
+                arn:aws:s3:::${CloudTrailBucket}
+          - 
+            Sid: "AWSCloudTrailWrite"
+            Effect: "Allow"
+            Principal: 
+              Service: "cloudtrail.amazonaws.com"
+            Action: "s3:PutObject"
+            Resource:
+              !Sub |-
+                arn:aws:s3:::${CloudTrailBucket}/AWSLogs/${AWS::AccountId}/*
+            Condition: 
+              StringEquals:
+                s3:x-amz-acl: "bucket-owner-full-control"
+
+  LogGroup:
+    Type: AWS::Logs::LogGroup
+    Properties:
+      RetentionInDays: 365
+
+  CloudTrailLogsRole:
+    Type: AWS::IAM::Role
+    Properties:
+      RoleName: !Join ['-', ['swb', 'initial-stack', 'cloudtrail-role']]
+      AssumeRolePolicyDocument:
+        Statement:
+        - Action: sts:AssumeRole
```

Comment:
Since it's part of the `AssumeRolePolicyDocument`, the resource is this role itself.

Reply:
Only the principal linked to this action (`cloudtrail.amazon.aws.com`) can assume this role to perform its actions. This can be seen under the role's trust relationship tab.